### PR TITLE
Add support for package pinning.

### PR DESCRIPTION
**Pull request by jmpage**
Replaces: #1025.

Resolves: #743
Impacts: #674, #800

This change allows users to add additional package archives and pin packages as desired by customizing two variables in a new file called `prelude-pinned-packages.el`.

### CODE REVIEW

File: core/prelude-packages.el [hunk 1]
```diff
@@ -37,6 +37,15 @@
 
 (add-to-list 'package-archives
              '("melpa" . "https://melpa.org/packages/") t)
+
+;; load the pinned packages
+(let ((prelude-pinned-packages-file (expand-file-name "prelude-pinned-packages.el" prelude-dir)))
+  (if (file-exists-p prelude-pinned-packages-file)
+      (load prelude-pinned-packages-file)
+    (message "Missing pinned packages file %s" prelude-pinned-packages-file)
```
Comment (bbatsov) [inline core/prelude-packages.el]: I wouldn't warn if the file is missing. Users might interpret this as an indication something's wrong, when it actually isn't.

**Comment by bbatsov** [comment]
You'll have to rebase this PR.

File: core/prelude-packages.el (hunk 1 above)
Comment (jmpage) [inline core/prelude-packages.el]: I have addressed this in a new commit.

**Comment by jmpage** [comment]
@bbatsov Conflicts resolved.

**Pull request merged**